android/app/src/main/java/com/zulipmobile/sharing/SharingHelper.kt

```diff
@@ -75,26 +83,30 @@ private fun getParamsFromIntent(intent: Intent, contentResolver: ContentResolver
     //
     // it corresponds with `src/types.js#SharedData`.
     val params = Arguments.createMap()
-    when {
-        "text/plain" == intent.type -> {
-            val sharedText = intent.getStringExtra(Intent.EXTRA_TEXT)
-            params.putString("type", "text")
-            params.putString("sharedText", sharedText)
-        }
-        intent.type?.startsWith("image/") == true -> {
-            val url = intent.getParcelableExtra<Uri>(Intent.EXTRA_STREAM)
-                ?: throw ShareParamsParseException("Could not extract URL from Image Intent")
-            params.putString("type", "image")
-            params.putString("sharedImageUrl", url.toString())
-            params.putString("fileName", getFileName(url, contentResolver))
-        }
-        else -> {
-            val url = intent.getParcelableExtra<Uri>(Intent.EXTRA_STREAM)
-                ?: throw ShareParamsParseException("Could not extract URL from File Intent")
-            params.putString("type", "file")
-            params.putString("sharedFileUrl", url.toString())
-            params.putString("fileName", getFileName(url, contentResolver))
+    if ("text/plain" == intent.type) {
+        val sharedText = intent.getStringExtra(Intent.EXTRA_TEXT)
+        params.putString("type", "text")
+        params.putString("sharedText", sharedText)
+    } else {
+        val files = Arguments.createArray()
+        when (intent.action) {
+            Intent.ACTION_SEND -> {
+                val url = intent.getParcelableExtra<Uri>(Intent.EXTRA_STREAM)
+                    ?: throw ShareParamsParseException("Could not extract URL from File Intent")
+                val file = urlToSharedFile(url, contentResolver)
+                files.pushMap(file)
```

Comment:
Could tighten a bit by inlining `file` here -- the resulting line is only slightly longer.